main/src/modules/reactiveLayer/behaviorManager/src/tagging.cpp

```diff
@@ -11,7 +11,24 @@ void Tagging::run(Bottle args/*=Bottle()*/) {
     yInfo() << "Tagging::run";
     yDebug() << "send rpc to proactiveTagging";
     Bottle *sensation = sensation_port_in.read();
-    int id = yarp::os::Random::uniform(0, sensation->size()-1);
+    int id = 42;
+        
+    // If there are unknown agents, prioritise tagging it.
+    for (int i = 0; i < sensation->size(); i++)
+    {
+        string type = sensation->get(i).asList()->get(0).asString();
+        string name = sensation->get(i).asList()->get(1).asString();
+
+        if ((type == "agent") && (name == "partner"))
+        {
+            id = i;
+        }
+    }
+    if (id == 42)
+    {
+        id = yarp::os::Random::uniform(0, sensation->size() - 1);
+    }
+
```

Comment:
Can you set `id` to `id = yarp::os::Random::uniform(0, sensation->size() - 1);` by default, and just overwrite the id in case an unknown partner was found? I.e. can you move line 29 to line 14? This will make it conceptually more clear.

Thanks, Tobi